### ISSUE TITLE
raspberrypifw binaries do not work on raspberrypi3

### ISSUE THREAD

**Issue by makefu** [NixOS/nixpkgs#30026]
## Issue description

When trying to run any of the binaries packaged in raspberrypifw (more specifically i was looking for vcgencmd) on my raspberrypi3 i get:
```command
[nix-shell:/var/src/nixpkgs]$ vcgencmd 
bash: /nix/store/75kzz46fpnb37jv6r2pd8y7zdjg4l5mg-raspberrypi-firmware-1.20170811/bin/vcgencmd: Accessing a corrupted shared library
```

### Steps to reproduce
```
nix-shell -p raspberrypifw --run vcgencmd
```

## Technical details

* System: 18.03pre-git (Impala)
* Nix version: nix-env (Nix) 1.11.14
* Nixpkgs version: 18.03pre-git, rev: 799435b7cab97a39893a104999b3bc589e1172b1
* Sandboxing enabled: true

**Comment by Mic92** [comment]
The package does patchelf on 32-bit arm executable and set a 64-bit link-loader (on aarch64):

```
$ file vcgencmd
vcgencmd: ELF 32-bit LSB executable, ARM, EABI5 version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux-armhf.so.3, for GNU/Linux 2.6.32, not stripped
```

**Comment by dezgeg** [comment]
No, we don't. AFAIK it's more difficult than the equivalent on x86 since there's no equivalent multilib support for that in GCC (i.e. no `gcc -m32`) so an AArch64 -> ARM cross compiler is needed, I believe.

**Comment by dezgeg** [comment]
I changed the `raspberrypifw` package to only contain the firmware blobs and added a separate package `raspberrypi-tools` which builds the userland tools from the BSD3 source.

I tested that it compiles on AArch64, but don't have the hardware on hand right now (only RPi 1) so please test and close if it works now.

**Comment by dezgeg** [comment]
On IRC I got a test report that the `vchiq` module isn't working out-of-the-box on mainline, but there are kernel patches floating/merged that will make it work on either 4.17 or 4.18.

**Comment by florianjacob** [comment]
This seems to work now as of 18.09 and kernel 4.18.16:

```
$ sudo vcgencmd measure_temp
temp=37.6'C
```

Sidenote: Requires root access. Without, this fails with a misleading error message:
```
$ vcgencmd
VCHI initialization failed
```
